llama_stack/providers/utils/scoring/base_scoring_fn.py

```diff
@@ -13,12 +13,51 @@
 
 class BaseScoringFn(ABC):
```

Comment:
I still don't understand why we have these base classes because we have already declared what our impls need to do in terms of datatypes in our APIs. so the datatype for a scoring function already exists. Let's say I am implementing a new scoring function -- why do I need another base class and inherit from there? If there is some utilities I need for implementing the functions, they would be just utils / free functions or in the worst case, some mixins. 

Can you explain the need for base classes please? I am very allergic to inheritance as you and @raghotham knows :)

Reply:
This `class BaseScoringFn(ABC):` is mostly separated out based on feedback from Tejas for his use case without 
 registration (syncing with you offline). 

For our llama-stack implementations, I agree we don't need this separate `BaseScoringFn`, and could just use `RegisteredBaseScoringFn` as mixins.

Reply:
@yanxi0830 I see, interesting. Thanks for the explanation. We can keep this as is for now.